homeassistant/components/ohme/entity.py

```diff
@@ -0,0 +1,38 @@
+"""Base class for entities."""
+
+from ohme import OhmeApiClient
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import OhmeCoordinator
+
+
+class OhmeEntity(CoordinatorEntity[OhmeCoordinator]):
+    """Base class for all Ohme entities."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: OhmeCoordinator,
+        hass: HomeAssistant,
+        client: OhmeApiClient,
+        entity_description: EntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+
+        self.entity_description = entity_description
+        self._attr_translation_key = entity_description.key
+        self._attr_device_info = DeviceInfo(**client.get_device_info())
```

Comment:
not a big fan of this tbh. I'd rather see here what the device is going to look like

Reply:
The current approach is that the API library returns a dictionary with the same keys as needed for DeviceInfo. Would you just explicitly pull each property here or take a different approach?

Reply:
Actually yes, I think so. Otherwise we'd need to check the library to see what the DeviceInfo looks like and if there are ever changes made to its definition we need to touch the library.